### PR TITLE
Enginko

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/enginko/MCF-LW12CO2E/uplink.js [hunk 1]
```diff
@@ -0,0 +1,912 @@
...
+      topic = "co2";
+      break;
+    default:
+      content = null;
+      break;
+  }
+
+  for (let i = 0; i < content.length; i++) {
+    let { value } = content[i];
+    if (isItNumber(value)) {
```
Comment (mariusgiger) [inline types/enginko/MCF-LW12CO2E/uplink.js:906]: rename to isNumeric

File: types/enginko/MCF-LW12CO2E/uplink.js [hunk 2]
```diff
@@ -0,0 +1,912 @@
...
+    case "0B":
+      content = parseReportData(payload.trim());
+      topic = "report";
+      break;
+    case "0C":
+      content = parseVOC(payload.trim(), false); // false is for new sensor
+      topic = "voc";
+      break;
+    case "0E":
+      content = parseCo2(payload.trim(), false); // false is for new sensor
```
Comment (mariusgiger) [inline types/enginko/MCF-LW12CO2E/uplink.js:896]: what does isNewSensor mean?

**Review by mariusgiger — CHANGES_REQUESTED** [review]
please add schemas for all topics

File: types/enginko/MCF-LW12CO2E/co2.schema.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+  "$id": "https://akenza.io/dragino/lse01/default.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Default",
+  "type": "object",
+  "properties": {
+    "rfu": {
+      "title": "RFU",
+      "description": "RFU",
```
[the remark this answers is not in the document]
Comment (J2thatsme) [reply types/enginko/MCF-LW12CO2E/co2.schema.json:9]: idk gonna delete it

[the remark this answers is not in the document]
Comment (mariusgiger) [reply types/enginko/MCF-LW12CO2E/co2.schema.json:9]: rather find it our :D

File: types/enginko/MCF-LW12CO2E/meta.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "Enginko",
+  "version": "1.0.0",
+  "manufacturer": "MCF-LW12CO2E",
+  "url": "https://enginko.com/en/solutions/mcf-lw12co2/",
+  "description": "This device reads temperature, humidity, pressure, light intensity, air quality (expressed as IAQ or bVOC value) and CO2 and sends collected data over the LoRaWAN® network. 3 traffic light LEDs for air quality display with adjustable brightness. Ideally suited for a wide range of applications such as building automation, condition monitoring, predictive maintenance, security, healthcare and many more use cases.",
```
Comment (sevyakenza) [inline types/enginko/MCF-LW12CO2E/meta.json:6]: @J2thatsme please use the description defined in the task:  This device reads temperature, humidity, pressure, light intensity, air quality (expressed as IAQ or bVOC value) and CO2. 3 traffic light LEDs for air quality display with adjustable brightness. Ideally suited for a wide range of applications such as building automation, condition monitoring, predictive maintenance, security, healthcare and many more use cases.

File: types/enginko/MCF-LW12CO2E/co2.schema.json [hunk 5]
```diff
@@ -0,0 +1,72 @@
+{
+  "$id": "https://akenza.io/dragino/lse01/default.schema.json",
```
Comment (sevyakenza) [inline types/enginko/MCF-LW12CO2E/co2.schema.json:2]: @J2thatsme this seems to be wrong

Comment (J2thatsme) [reply types/enginko/MCF-LW12CO2E/co2.schema.json:2]: fixed

File: types/enginko/MCF-LW12CO2E/uplink.js (hunk 2 above)
Comment (J2thatsme) [reply types/enginko/MCF-LW12CO2E/uplink.js:896]: looks like different encryption

**Review by mariusgiger — APPROVED** [review]

File: types/enginko/MCF-LW12CO2E/uplink.js [hunk 6]
```diff
@@ -0,0 +1,917 @@
...
+      variable: "batteryLevel",
+      value: Number(parseInt(payload.substring(62, 64), 16).toFixed()) || 0,
+      unit: "%",
+    };
+    const rfu = {
+      variable: "rfu",
+      value: payload.substring(64),
+    };
+
+    // TODO why are the measurements converted to one object? this will lead to duplicated keys and eventually only contain the latest measurement
```
Comment (mariusgiger) [inline types/enginko/MCF-LW12CO2E/uplink.js:537]: @J2thatsme please plan a fix for this

**Pull request merged**